orders/views.py

```diff
@@ -1,3 +1,99 @@
-from django.shortcuts import render
+import json
+
+from django.http       import JsonResponse
+from django.views      import View
+
+from decorators        import validate_login
+from orders.models     import Order, Status, ScheduleOrder
+from taxis.models      import Schedule
+
+STATUS_BOOK     = 1
+STATUS_FINISHED = 2
+STATUS_CANCLED  = 3
+TAX             = 8800
+
+class BookView(View):
+    @validate_login
+    def post(self, request):
+        try:
+            data               = json.loads(request.body)
+            passenger_number   = data.get('passenger_number', None)
+            going_schedule_id  = data.get('going_schedule_id', None)
+            coming_schedule_id = data.get('coming_schedule_id', None)
+            user               = request.user
+            
+            schedule_ids = [ schedule_id for schedule_id in (going_schedule_id, coming_schedule_id) if schedule_id ]
+
+            for schedule_id in schedule_ids:
+                if passenger_number > Schedule.objects.get(id=schedule_id).seat_remain:
+                    return JsonResponse({'message': 'no_seat_remain'}, status=401)
+
+            order = Order.objects.create(user=user, status_id=STATUS_BOOK)
+
+            for schedule_id in schedule_ids:
+                schedule_orders = ScheduleOrder.objects.create(
+                        order            = order,
+                        schedule_id      = schedule_id,
+                        tax              = TAX,
+                        passenger_number = passenger_number
+                        ) 
+
+                schedule  = Schedule.objects.get(id=schedule_id)
+                schedule.seat_remain -= passenger_number
+                schedule.save()
+
+            return JsonResponse({'message': 'success'}, status=201)
+        except json.decoder.JSONDecodeError:
+            return JsonResponse({'message': 'no_body'}, status=400)
+        except KeyError:
+            return JsonResponse({'message': 'key_error'}, status=400)
+        except Schedule.DoesNotExist:
+            return JsonResponse({'message': 'invalid_id'}, status=400)
+
+    @validate_login
+    def get(self, request):
+        user   = request.user
+        orders = Order.objects.filter(user=user).prefetch_related('scheduleorder_set__schedule__course').order_by('-created_at')
+        
+        results = [
+                {
+                    'order_id'                : order.id,
+                    'round_trip'              : '왕복' if order.scheduleorder_set.count() == 2 else '편도',
+                    'departure_location'      : order.scheduleorder_set.all()[0].schedule.course.departure_location.name,
```

Comment:
```suggestion
                    'departure_location'      : order.scheduleorder_set.first().schedule.course.departure_location.name,
```
[0]을 사용하게 되면 데이터가 없었을 때 index 에러가 나올 수 있습니다.

Reply:
order가 있으면 schedule_order가 무조건 있어야하는 로직입니다.
그래도 혹시 모르니 IndexError 에러 예외처리 해놓도록 하겠습니다.
그냥 first()를 이용할 경우에 order에 대응하는 schedule_order가 없다면 None이 반환되고 None.schedule... 이 되어 오류가 발생할 것으로 생각됩니다.